src/common/non_file_printing_counter.cpp

```diff
@@ -0,0 +1,26 @@
+/**
+ * @file non_file_printing_counter.cpp
+ * @author Radek Vana
+ * @date 2021-04-29
+ */
+
+#include "non_file_printing_counter.hpp"
+#include "rtos_api.hpp"
+
+unsigned NonFilePrintingCounter::counter = 0;
+
+void NonFilePrintingCounter::increment() {
+    CriticalSection C;
```

Comment:
Is the critical section really necessary? How about std::atomic? The MCU should have atomic instruction support.